### PR TITLE
add cramer's v for categorical column

### PR DESCRIPTION
**Pull request by jinglinpeng**
# Description

add cramer's v correlation for categorical column.
# How Has This Been Tested?
add unit test

# Snapshots:
stat in table:
Screen Shot 2021-03-01 at 11 38 00 PM

cramer's v tab:
Screen Shot 2021-03-01 at 11 37 53 PM

### CODE REVIEW

File: dataprep/eda/correlation/compute/overview.py [hunk 1]
```diff
@@ -24,132 +26,162 @@ def _calc_overview(
     value_range: Optional[Tuple[float, float]] = None,
     k: Optional[int] = None,
 ) -> Intermediate:
-    # pylint: disable=too-many-statements,too-many-locals,too-many-branches
+    # pylint: disable=too-many-statements,too-many-locals,too-many-branches,too-many-boolean-expressions
```
Comment (dovahcrow) [inline dataprep/eda/correlation/compute/overview.py:29]: Can we break this function down and remove these disables as much as possible?

File: dataprep/eda/correlation/compute/overview.py [hunk 2]
```diff
@@ -196,14 +232,67 @@ def _spearman_nxn(df: DataArray) -> da.Array:
...
+    Calculate column-wise kendalltau correlation.
+    """
     return (
         df.frame.repartition(npartitions=1)
         .map_partitions(partial(pd.DataFrame.corr, method="kendall"))
         .to_dask_array()
     )
 
 
+def _cramerv_nxn(df: DataArray) -> da.Array:
```
Comment (dovahcrow) [inline dataprep/eda/correlation/compute/overview.py:245]: What's the performance / complexity of this function? Is cramerv disabled by default?

File: dataprep/eda/correlation/compute/overview.py [hunk 3]
```diff
@@ -24,132 +26,162 @@ def _calc_overview(
...
+    # columns: used column names
+    # cordx, cordy: used to locate a cell in correlation matrix.
+    # corrs: correlation matrix.
+    method2columns: Dict[CorrelationMethod, pd.Index] = {}
+    method2cordx: Dict[CorrelationMethod, np.ndarray] = {}
+    method2cordy: Dict[CorrelationMethod, np.ndarray] = {}
+    method2corrs: Dict[CorrelationMethod, da.Array] = {}
+
+    if contains_num and (cfg.pearson.enable or cfg.stats.enable):
+        method2columns[CorrelationMethod.Pearson] = num_df.columns
```
[the remark this answers is not in the document]
Comment (jinglinpeng) [reply dataprep/eda/correlation/compute/overview.py:63]: cramer's v correlation only applies to categorical columns. Hence I distinguish `num_df` and `cat_df`.

[the remark this answers is not in the document]
Comment (dovahcrow) [reply dataprep/eda/correlation/compute/overview.py:63]: so the cords are also different? How about breaking this down into two functions, one for cat and one for num? I feel a function with more than 100 lines is a little bit long.

File: dataprep/eda/correlation/compute/overview.py (hunk 2 above)
Comment (jinglinpeng) [reply dataprep/eda/correlation/compute/overview.py:245]: Haven't test the performance. It is enabled by default. I think it's better to have both categorical correlation and numerical correlation for the overview (we may just choose one numerical correlation such as person correlation for efficiency). Or another choice is to enable Cramerv for some data (e.g., when number of columns is small).  Cramerv should be expensive, around `n^2*v^2`, where `n` is the number of columns and `v` is the distinct values.  The code of this part is similar to pandas-profiling.

File: dataprep/eda/correlation/compute/overview.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jinglinpeng) [reply dataprep/eda/correlation/compute/overview.py:63]: Yeah. cords and columns are different. For now there only exists two scenarios: some correlations use categorical columns and others use numerical columns. I also considered we may also want to support other correlations that may be different from these two cases (e.g., phik use all columns), hence I just write each correlation one by one. 

I can break down the function to make each part smaller.

File: dataprep/eda/correlation/compute/overview.py (hunk 2 above)
Comment (dovahcrow) [reply dataprep/eda/correlation/compute/overview.py:245]: how about have a try on the titanic and if it is very slow then we disable it by default? We can also discuss in the meeting if we can have the number before the meeting.

**Pull request closed without merging**